### PR TITLE
[Bugfix:Calendar] Check if array null in CalendarView.php

### PR DESCRIPTION
**Pull request by shreyashrpawar**
Solved issue  #10207

The issue is generating because we are taking a value of first index of an array which is empty. So, I have added nested ternary block which will take that first element if the length of an array is greater than 0.

### What is the current behavior?
Now, because of the modification it is working fine

### CODE REVIEW

**Review by michaelkrar — CHANGES_REQUESTED** [review]
Not an issue in firefox. Was able to replicate the issue on chrome. The changes completely fix the issue and are non-breaking. Changes are a reasonable solution and the code is formatted well / logic is simple to follow. Also changes appear to not break anything in Firefox as well as student/ta logins.
PR Title shouuld be changed to be more specific as to what bug is being fixed, and also include the prefix: [Bugfix:Calendar]

**Review by GO-viper7 — CHANGES_REQUESTED** [review]
PR title should be bit more concise, as it is limited to 40 characters excluding [Bugfix:Calendar] and gives title-check fail if it is not followed.

**Review by shreyashrpawar — COMMENTED** [review]
Simplified the code as instructed by [arougey](https://github.com/arougey).
Also changed the title as suggested by [GO-viper7](https://github.com/GO-viper7).

**Comment by IDzyre** [comment]
All of these warnings cause misalligned UI, it would be better to state what the warning is for, and/or where it is found, like `Fix Superuser Calendar View` or `Check if calendar array is empty`

**Comment by shreyashrpawar** [comment]
> All of these warnings cause misalligned UI, it would be better to state what the warning is for, and/or where it is found, like `Fix Superuser Calendar View` or `Check if calendar array is empty`

Thanks for the suggestion. I have changed the title as more precise. Kindly approve. [IDzyre](https://github.com/IDzyre)   [GO-viper7](https://github.com/GO-viper7)   [arougey](https://github.com/arougey)   [michaelkrar](https://github.com/michaelkrar)

**Review by iamdanielyun — APPROVED** [review]

**Review by JoJo-ESC — APPROVED** [review]
Code looks good and changes can be seen on my machine. If I had any common it would be to pls change the title to calendar_fix.

File: site/app/views/calendar/CalendarView.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ public function showCalendar(CalendarInfo $info, array $courses, bool $in_course
             "show_legend" => $show_legend,
             "color_options" => $course_colors,
             "show_all_cookie" => isset($_COOKIE['calendar_show_all']) ? $_COOKIE['calendar_show_all'] : 1,
-            "calendar_course_cookie" => isset($_COOKIE['calendar_course']) ? $_COOKIE['calendar_course'] : $formatted_courses[0],
+            "calendar_course_cookie" => $_COOKIE['calendar_course'] ?? ($formatted_courses[0] ?? NULL),
```
Comment (IDzyre) [inline site/app/views/calendar/CalendarView.php:105]:
```suggestion
            "calendar_course_cookie" => $_COOKIE['calendar_course'] ?? ($formatted_courses[0] ?? null),
```

for php lint

**Review by IDzyre — CHANGES_REQUESTED** [review]

**Review by IDzyre — APPROVED** [review]

**Review by cjreed121 — APPROVED** [review]
Looks good to me

**Pull request merged**